examples/integrations/gcp/src/coordinator/pipelines/text/ner.py

```diff
@@ -68,20 +73,118 @@ def run(self, training_file_uri: str, job_name: str) -> JobStatus:
+                prediction = prediction_data['prediction']
+
+                if not len(prediction):
+                    # No prediction..
+                    continue
+
+                for name, start, end in zip(
+                        prediction['displayNames'],
+                        prediction['textSegmentStartOffsets'],
+                        prediction['textSegmentEndOffsets']):
+                    start = int(start)  # Start is inclusive
+                    end = int(
+                        end
+                    ) + 1  # vertex end is inclusive. We want exclusive for labelbox
+                    yield {
+                        'uuid': str(uuid.uuid4()),
+                        'dataRow': {
+                            'id': data_row_id
+                        },
+                        'schemaId': tools[name],
+                        'location': {
+                            'start': start,
+                            'end': end
+                        }
+                    }
+
+    def run(self, etl_file: str, model_run_id: str, model: Model,
+            job_name: str):
+        batch_prediction_job = self.batch_predict(etl_file, model, job_name,
+                                                  "ner")
+        model_run = self.lb_client._get_single(ModelRun, model_run_id)
+        tools = self.get_tool_info(model_run.model_id)
+        annotation_data = list(
+            self.process_predictions(batch_prediction_job, tools))
+        predictions = list(NDJsonConverter.deserialize(annotation_data))
+        labels = self.export_model_run_labels(model_run_id, 'text')
+        self.compute_metrics(labels, predictions, tools)
+        upload_task = model_run.add_predictions(
+            f'diagnostics-import-{uuid.uuid4()}',
+            NDJsonConverter.serialize(predictions))
+        upload_task.wait_until_done()
+        # TODO: Report errors without necessarily breaking the job...
+        logger.info(
+            f"IMPORT ERRORS : {upload_task.errors}. Imported {len(predictions)}"
+        )
+        return JobStatus(JobState.SUCCESS)
+
+    def compute_metrics(self, labels, predictions, tools):
+        tool_name_lookup = {v: k for k, v in tools.items()}
+        pairs = get_label_pairs(labels, predictions, filter_mismatch=True)
+        for (ground_truth, prediction) in pairs.values():
+            # Assign names instead of feature ids for classes.
+            for annotation in prediction.annotations:
+                # Unused for classifications so actually doesn't matter.
+                # Only needed so that all pairs have names.
+                if isinstance(annotation.value, TextEntity):
+                    annotation.name = tool_name_lookup[
+                        annotation.feature_schema_id].replace(' ', '-')
+            # TODO: Metrics are not yet supported for TextEntity
+            #metrics = []
+            #prediction.annotations.extend(metrics)
```

Comment:
Jtso is working on a separate ticket to add this support to the SDK. Part of that ticket is to follow up here.